From 02fa8f2f0f474db4a797dd67730e033ea9c4d658 Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Mon, 27 Jun 2016 12:03:55 +0200
Subject: [PATCH] fcoe/cleanup-fcoe.sh: don't copy shm files to state

with systemd version 230, this is not necessary anymore
systemd commit cacf980ed44a28e276a6cc7f8fc41f991e2ab354
because /dev/shm is selinux relabled.
---
 modules.d/95fcoe/cleanup-fcoe.sh | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/modules.d/95fcoe/cleanup-fcoe.sh b/modules.d/95fcoe/cleanup-fcoe.sh
index 5ff4d054..1872156e 100644
--- a/modules.d/95fcoe/cleanup-fcoe.sh
+++ b/modules.d/95fcoe/cleanup-fcoe.sh
@@ -4,7 +4,11 @@
 
 if [ -e /var/run/lldpad.pid ]; then
     lldpad -k
-    mkdir -m 0755 -p /run/initramfs/state/dev/shm
-    cp /dev/shm/lldpad.state /run/initramfs/state/dev/shm/ > /dev/null 2>&1
-    echo "files /dev/shm/lldpad.state" >> /run/initramfs/rwtab
+    # with systemd version 230, this is not necessary anymore
+    # systemd commit cacf980ed44a28e276a6cc7f8fc41f991e2ab354
+    if [ -z "$DRACUT_SYSTEMD" ]; then
+        mkdir -m 0755 -p /run/initramfs/state/dev/shm
+        cp /dev/shm/lldpad.state /run/initramfs/state/dev/shm/ > /dev/null 2>&1
+        echo "files /dev/shm/lldpad.state" >> /run/initramfs/rwtab
+    fi
 fi
